src/enums/variant-payloads.md

```diff
@@ -9,8 +9,15 @@ You can define richer enums where the variants carry data. You can then use the
 
 <details>
 
+Key Points:
+  
+* This might be the first time seeing pattern matching so perhaps introduce the topic, more details are provided in further slides.
+  * Match is very similar to checking against an if expression. The difference is an if expression returns a boolean value, where this match expression can return any type.
```

Comment:
I'm not sure what you mean by this. The match expression as a whole can return any type, but so can an if expression. The match arms must have patterns matching the type of the expression that is being matched on.

Reply:
I meant compared to if expressions in other languages, but I can see how this is confusing and removed the line.